alt_e2eshark/e2e_testing/registry.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+from e2e_testing.framework import Test
+
+GLOBAL_TEST_LIST = []
+
+_SEEN_NAMES = set()
+
+
+def register_test(test_class: type, test_name: str):
```

Comment:
might be a good idea to add a 

```
def e2e_shark_test_case(test_name: str):
   return lambda test_class: register_test(test_class, test_name)
```

This will allow us to do
```
@e2e_shark_test_case("LLAMA3")
class Llama3TestCase:
  ...
```
with a decorator up front